resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/PVTestsSuite.scala

```diff
@@ -42,23 +48,14 @@ private[spark] trait PVTestsSuite { k8sSuite: KubernetesSuite =>
       .withKind("PersistentVolume")
       .withApiVersion("v1")
       .withNewMetadata()
-        .withName("test-local-pv")
+        .withName(PV_NAME)
       .endMetadata()
       .withNewSpec()
         .withCapacity(Map("storage" -> new QuantityBuilder().withAmount("1Gi").build()).asJava)
         .withAccessModes("ReadWriteOnce")
         .withPersistentVolumeReclaimPolicy("Retain")
-        .withStorageClassName("test-local-storage")
-        .withLocal(new LocalVolumeSourceBuilder().withPath(VM_PATH).build())
-          .withNewNodeAffinity()
-            .withNewRequired()
-              .withNodeSelectorTerms(new NodeSelectorTermBuilder()
-                .withMatchExpressions(new NodeSelectorRequirementBuilder()
-                  .withKey("kubernetes.io/hostname")
-                  .withOperator("In")
-                  .withValues("minikube", "docker-for-desktop", "docker-desktop").build()).build())
```

Comment:
Hi, @holdenk . Could you confirm that this doesn't break the other backends?

Reply:
So I thought this test only ran on MiniKube already based on the `test("PVs with local storage", k8sTestTag, MinikubeTag`

Reply:
I'm saying this test is already excluded from non-minikube tests so I'm not adding an exclusion. But I'm happy to leave an JIRA ( https://issues.apache.org/jira/browse/SPARK-29076 ) to explore what would be needed to remove the minikube only restriction for this test. I don't think it belongs as a TODO in the code though. I

Reply:
It seems that I wasn't clear enough.

What I meant in the above is that the following is wrong.
> I'm saying this test is already excluded from non-minikube tests 

The following is the result of `DockerDesktop` backend on the master branch as of today.
Please try in your environment.

```
$ kubectl cluster-info
Kubernetes master is running at https://kubernetes.docker.internal:6443
KubeDNS is running at https://kubernetes.docker.internal:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

$ ./dev/make-distribution.sh --pip --tgz -Phadoop-2.7 -Pkubernetes

$ resource-managers/kubernetes/integration-tests/dev/dev-run-integration-tests.sh --deploy-mode docker-for-desktop --spark-tgz $PWD/spark-*.tgz
Run starting. Expected test count is: 18
KubernetesSuite:
- Run SparkPi with no resources
- Run SparkPi with a very long application name.
- Use SparkLauncher.NO_RESOURCE
- Run SparkPi with a master URL without a scheme.
- Run SparkPi with an argument.
- Run SparkPi with custom labels, annotations, and environment variables.
- All pods have the same service account by default
- Run extraJVMOptions check on driver
- Run SparkRemoteFileTest using a remote data file
- Run SparkPi with env and mount secrets.
- Run PySpark on simple pi.py example
- Run PySpark with Python2 to test a pyfiles example
- Run PySpark with Python3 to test a pyfiles example
- Run PySpark with memory customization
- Run in client mode.
- Start pod creation from template
- PVs with local storage
- Launcher client dependencies *** FAILED ***
Tests: succeeded 17, failed 1, canceled 0, ignored 0, pending 0
*** 1 TEST FAILED ***
```

I didn't test this on this PR. Please try to run it. That was my initial request, @holdenk .

Reply:
Note that `- PVs with local storage` test passed as of today.

Reply:
Gotcha. Would it make sense (in the follow up) to add a test tag for able to run on docker4desktop or minikube but not in fully distributed so folks can still run the integration tests with the script?
I'll work around the possible d4d issue.

Reply:
I did some more poking and running docker-for-desktop is kind of a pain for me, 0% of the tests pass with the default configuration. This test is already tagged for minikube only, so I'd argue the correct run path for running the docker-for-desktop-tests would be `resource-managers/kubernetes/integration-tests/dev/dev-run-integration-tests.sh --deploy-mode docker-for-desktop --spark-tgz $PWD/spark-*.tgz --exclude-tags minikube` and then the docker for desktop test suite passes.

I'm down to spend a few hours trying to get docker for desktop working on my machine if folks actually use this for their testing rather than MiniKube, but I think given that the https://github.com/apache/spark/blob/master/resource-managers/kubernetes/integration-tests/README.md suggests `MiniKube` as the way to run our integration tests and the test is already tagged for MiniKube only, just supporting MiniKube and leaving a follow up issue for someone who cares about docker-4-desktop is an OK path forward.

If that is not OK I could switch this into a README PR that tells people to set up the bind mounts manually, would that be an ok option with you?

Reply:
Yep. The following was my suggestion (https://github.com/apache/spark/pull/25748#discussion_r323895070), too. Let's have IDed TODO comment and move forward~
>  leaving a follow up issue for someone who cares about docker-4-desktop is an OK path forward.